server/compaction-coordinator/src/main/java/org/apache/accumulo/coordinator/DeadCompactionDetector.java

```diff
@@ -85,31 +103,55 @@ private void detectDeadCompactions() {
 
     running.forEach((ecid) -> {
       if (tabletCompactions.remove(ecid) != null) {
-        log.trace("Removed {} running on a compactor", ecid);
+        log.trace("Removed compaction {} running on a compactor", ecid);
+      }
+      if (this.deadCompactions.remove(ecid) != null) {
+        log.trace("Removed {} from the dead compaction map, it's running on a compactor", ecid);
       }
     });
 
     // Determine which compactions are currently committing and remove those
     context.getAmple().getExternalCompactionFinalStates()
-        .map(ecfs -> ecfs.getExternalCompactionId()).forEach(tabletCompactions::remove);
+        .map(ecfs -> ecfs.getExternalCompactionId()).forEach(ecid -> {
+          if (tabletCompactions.remove(ecid) != null) {
+            log.trace("Removed compaction {} that is committing", ecid);
+          }
+          if (this.deadCompactions.remove(ecid) != null) {
+            log.trace("Removed {} from the dead compaction map, it's committing", ecid);
+          }
+        });
 
-    tabletCompactions
-        .forEach((ecid, extent) -> log.debug("Detected dead compaction {} {}", ecid, extent));
+    tabletCompactions.forEach((ecid, extent) -> {
+      log.debug("Possible dead compaction detected {} {}", ecid, extent);
+      this.deadCompactions.putIfAbsent(ecid, System.currentTimeMillis());
```

Comment:
Using a count instead of time may simplify code.  Not sure if the following is correct, I think it may increment a count or set it to 1 if it does not exists.

```suggestion
      this.deadCompactions.merge(ecid, 1, Long::sum);
```

Reply:
It looks like it will set it to 1 if  it doesn't exist, I'm not sure if it will increment or not based on the javadoc.

Reply:
It works.

Reply:
I applied this change in 95e8fec149722ac3f205734103223e3f42b33438